doc/semantics/CloudHaskellSemantics.tex

```diff
@@ -700,6 +700,21 @@ \subsection{Ordering and Typed Channels}
 for all messages from $P$ to $Q$, rather than using an ordered connection per
 typed channel plus one for direct messages.
 
+\subsection{Registry}
+
+For uniformity in Cloud Haskell can accept both local and remote processes
+names can be stored in Registry. Registering a local process at a remote
+node is suppoerted to using operation \texttt{registerRemoveAsync}.
+As a consequence, when a message is sent to a remote node using \texttt{nsendRemote}
+there is no guarantee that the process that should receive the message is
+located at node; thus it may be necessary to relay the message to a process
+on yet another node.
+
+Both `nsend` and `nsendRemote` operations are asynchronous that doesn't
+check existance of the target process, if process died then message will
+be silently discarded.
```

Comment:
This text needs to be edited. I offer a version below. However a few things are amiss.

* Nothing is said about ordering of messages sent with nsend and nsendRemote
* Nothing is said about how the registry behaves on failures.
* The subsection is included in Open Issues, but I don't see any issue remarked as open.

It is apparent now that the semantics need to include `register`, `unregister` and possibly `nsendRemote` as primitives.

---
The identifiers of both local and remote processes can be stored in the Registry. The operation \texttt{registerRemoteAsync} can register processes at remote nodes. When a message is sent to a remote node using \texttt{nsendRemote} there is no guarantee that the process that should receive the message is located at the node; thus it may be necessary to relay the message to a process on yet another node.

Both operations `nsend` and `nsendRemote` discard the messages if no process is registered with the given name.